social-network-data-migration-cli/src/main/java/pl/edu/agh/socialnetworkdatamigration/cli/SocialNetworkDataMigrationCLI.java

```diff
@@ -1,88 +1,153 @@
-        var schemaMapping = mappingLoader.loadFromJson(jsonStr);
+        Option mapping = Option.builder()
+                .hasArg(true)
+                .argName("Path to mapping file")
+                .required(true)
+                .longOpt("mapping")
+                .build();
 
-        try (var migrator = new CSVMigrator(configPath, csvDataPath, withHeaders)) {
-            migrator.migrateData(schemaMapping);
-        }
-    }
+        Option noHeaders = Option.builder()
+                .hasArg(false)
+                .required(false)
+                .longOpt("no-headers")
+                .build();
 
-    private static void interactiveSQL(String[] args) throws IOException {
-        var creator = new InteractiveSQLMappingCreator(args[1]);
-        SchemaMapping mapping = creator.createInteractively();
-        System.out.println(mapping);
-    }
+        Option merge = Option.builder()
+                .hasArg(false)
+                .required(false)
+                .longOpt("merge")
+                .build();
 
-    private static void interactiveCSV(String[] args) throws Exception {
-        boolean withHeaders = args.length == 6 && args[4].equals("--no-headers");
-        InteractiveCSVMappingCreator mappingLoader = new InteractiveCSVMappingCreator(args[3], withHeaders);
-        CSVSchemaMapping mapping = mappingLoader.createInteractively();
-        try (var migrator = new CSVMigrator(args[2], args[3], withHeaders)) {
-            migrator.migrateData(mapping);
+        Option dryRun = Option.builder()
+                .hasArg(false)
+                .required(false)
+                .longOpt("dry-run")
+                .build();
+
+        Options options = new Options();
+        options.addOption(interactive).addOption(csv).addOption(config).addOption(mapping).
+                addOption(noHeaders).addOption(merge).addOption(dryRun);
+        CommandLineParser parser = new DefaultParser();
+        try {
+            return parser.parse(options, args);
+        } catch (ParseException e) {
+            System.out.println(e.getMessage());
+            printUsage();
+            System.exit(1);
+            return null;
         }
     }
 
     private static void printUsage() {
         System.out.println(
-                "Usage: java -jar SocialNetworkDataMigrationCLI <config-path> <mapping-path>\n" +
-                "       java -jar SocialNetworkDataMigrationCLI --csv <config-path> <data-path> <mapping-path> [--no-headers]\n" +
-                "       java -jar SocialNetworkDataMigrationCLI --i <config-path>\n" +
-                "       java -jar SocialNetworkDataMigrationCLI --csv --i <config-path> <data-path> [--no-headers]");
+                "Usage:\n" +
+                        "java -jar SocialNetworkDataMigrationCLI --config <config-path> --mapping <mapping-path> [-i] [--dry-run]" +
+                        " [--merge] [--csv <data-path> [--field-terminator <field-terminator>] [--no-headers]]\n"
```

Comment:
Mega propsy za użycie tego Commons CLI, dużo lepiej to teraz wygląda 👍🏼

Reply:
(chociaż akurat `--mapping` w tym usage powinno być opcjonalne, tak jak `[-i]`)

Reply:
Racja, --mapping i -i się wykluczają wzajemnie.